parsing/parser.mly

```diff
@@ -1935,12 +1939,17 @@ module_type:
   | STRUCT error
       { expecting $loc($1) "sig" }
   | FUNCTOR attrs = attributes args = functor_args
-    MINUSGREATER mty = module_type
+    MINUSGREATER mty = module_type mm = optional_at_mode_expr
       %prec below_WITH
       { wrap_mty_attrs ~loc:$sloc attrs (
-          List.fold_left (fun acc (startpos, arg) ->
-            mkmty ~loc:(startpos, $endpos) (Pmty_functor (arg, acc))
-          ) mty args
+          let mty, mm =
```

Comment:
I suggest a comment like "return modes go on the innermost functor arrow; all other return modes are empty."